cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/ConfiguratorTask.java

```diff
@@ -0,0 +1,124 @@
+
+  }
+
+  @Override
+  public void run(RunnableTaskContext context) throws Exception {
+    Injector injector = Guice.createInjector(new ConfiguratorTaskModule());
+    Configurator configurator = injector.getInstance(Configurator.class);
+    configurator.run(context);
+  }
+
+  /**
+   * Configurator performs the configurator config.
+   */
+  public static class Configurator {
+    private final Impersonator impersonator;
+    private final PluginFinder pluginFinder;
+    private final ArtifactRepository artifactRepository;
+    private final LocationFactory locationFactory;
+
+    @Inject
+    public Configurator(Impersonator impersonator, PluginFinder pluginFinder,
+                        ArtifactRepository artifactRepository, LocationFactory locationFactory) {
+      this.impersonator = impersonator;
+      this.pluginFinder = pluginFinder;
+      this.artifactRepository = artifactRepository;
+      this.locationFactory = locationFactory;
+    }
+
+    public void run(RunnableTaskContext context) throws Exception {
+      ConfiguratorConfig config = GSON.fromJson(context.getParam(), ConfiguratorConfig.class);
+
+      // Getting the pipeline app from appfabric
+      LOG.info(String.format("Fetching artifact '%s' from app-fabric to create artifact class loader.",
+                             config.getArtifactId().getName()));
+      Location artifactLocation = Locations.getLocationFromAbsolutePath(
+        locationFactory, config.getArtifactLocationURI().getPath());
+      OutputStream outputStream = artifactLocation.getOutputStream();
+      InputStream artifactBytes = artifactRepository.getArtifactBytes(config.getArtifactId());
+      ByteStreams.copy(artifactBytes, outputStream);
+      outputStream.close();
+      artifactBytes.close();
+      LOG.info(String.format("Successfully fetched artifact '%s'.", config.getArtifactId().getName()));
+
+
+      EntityImpersonator classLoaderImpersonator =
+        new EntityImpersonator(config.getArtifactId().toEntityId(), impersonator);
+      ClassLoader artifactClassLoader = artifactRepository.createArtifactClassLoader(artifactLocation,
+                                                                                     classLoaderImpersonator);
+
+      InMemoryConfigurator configurator = new InMemoryConfigurator(
+        config.getcConf(), config.getAppNamespace(), config.getArtifactId(),
+        config.getAppClassName(), pluginFinder,
+        artifactClassLoader,
+        config.getApplicationName(), config.getApplicationVersion(),
+        config.getConfigString());
+
+      ListenableFuture<ConfigResponse> future = configurator.config();
+      ConfigResponseResult result;
+
+      try {
+        result = new ConfigResponseResult(future.get(120, TimeUnit.SECONDS), null);
+      } catch (Exception ex) {
+        LOG.error("Encountered error while validating pipeline: ", ex);
+        result = new ConfigResponseResult(null, ex);
+      }
+
+      String json = GSON.toJson(result);
```

Comment:
You can't use GSON to serialize / deserialize exception object. You have to turn it to BasicThrowable.

Reply:
Done, exception is converted to BasicThrowable in `ConfigResponseResult`